integration/testrunner/http.go

```diff
@@ -64,18 +75,62 @@ func testHTTPEndpoint(ctx context.Context, tb testing.TB, endpointURL, idToken,
 	}); err != nil {
 		tb.Fatal(err)
 	}
-
 	bqClient := makeBigQueryClient(ctx, tb, projectID)
-
+	time.Sleep(10 * time.Second)
 	bqQuery := makeQueryForHTTP(*bqClient, id, projectID, datasetQuery)
-	queryIfAuditLogExistsWithRetries(ctx, tb, bqQuery, cfg, "httpEndpointTest")
+	tb.Log(bqQuery.Q)
+	results := queryIfAuditLogsExistsWithRetries(ctx, tb, bqQuery, cfg, "httpEndpointTest")
+	wantNum := 1
+	if len(results) != wantNum {
+		tb.Errorf("log number doesn't match (-want +got):\n - %d\n + %d\n", wantNum, len(results))
+	} else {
+		jsonPayloadInfo := parseJsonpayload(tb, results[0])
+		diff := diffResults(results[0], jsonPayloadInfo, getMode("HTTP"))
+		if diff != "" {
+			tb.Errorf(diff)
+		}
+	}
 }
 
 func makeQueryForHTTP(client bigquery.Client, id, projectID, datasetQuery string) *bigquery.Query {
-	// Cast to int64 because the result checker expects a number.
-	queryString := fmt.Sprintf("SELECT CAST(EXISTS (SELECT * FROM `%s.%s` WHERE labels.trace_id=?", projectID, datasetQuery)
-	queryString += ` AND jsonPayload.service_name IS NOT NULL`
-	queryString += ` AND jsonPayload.authentication_info.principal_email IS NOT NULL`
-	queryString += ") AS INT64)"
+	queryString := fmt.Sprintf(` WITH temptable AS (
+		SELECT *
+		FROM %s.%s
+		WHERE labels.trace_id=?
+ 	)
+ 	SELECT TO_JSON(t) as result FROM temptable as t
+	`, projectID, datasetQuery)
 	return makeQuery(client, id, queryString)
 }
+
+func diffResults(logEntry *logpb.LogEntry, jsonPayloadInfo *audit.AuditLog, isHTTPService bool) string {
```

Comment:
If we're going to do a custom differ (I'm in*differ*ent), we should just log errors here instead of returning a diff. For example: https://github.com/abcxyz/jvs/blob/4312d969602a3002769cb0806023f920f0b87477/pkg/cli/token_test.go#L140-L155

Reply:
Done